docs/source/configuration/hardware.rst

```diff
@@ -83,3 +86,115 @@ You must now specify which GPIO platform pilight is running on. Refer to the set
 
 The comport value needs to correspond to a valid COM device on your platform. On Windows this value is generally formatted as COM1, on Linux as /dev/ttyUSB0, and on FreeBSD /dev/cuau0.
 
+.. versionadded:: nightly
+
+MQTT
+----
+
+The hardware module is doing all the mqtt protocol stuff required to establish mqtt connections and subscriptions and publishing and receiving of mqtt messages.
+
+When pilight starts, the module will automatically be initialized and then clients can subscribe for messages published by the broker and/or publish messages themselves.
+When an attempt is made to subscribe or publish for the first time, the hardware module creates a socket, makes a network connection and does a CONNECT request and, for subscribe the mqtt module also does a SUBSCRIBE request. The client is added to the client pool and the client is informed about the status via a callback. The clientid is the unique key to the clients data in the client pool. Therefor the clientids must be unique.
+
+Each client can individulally specify a keepalive time and the established connection and/or subscription will be kept alive at least for the given time. If the keepalive time is set to ``0`` (zero), the connection and/or subscription will be kept alive "forever" and will be used for new publish request and reception of messages.
+
+If a connection cannot be established or dies for whatever reason, the affected client(s) will immediately be notified and then can try to subscribe or publish again.
+
+Currently three mqtt client types are available for pilight:
+
+- An mqtt_switch protocol that can be used to control external mqtt enabled devices and can handle feedback from those devices. Connection and or subscription are kept alive "forever".
+- An mqtt_publish action that can be used to simply publish mqtt messages from rules. Connection is kept alive just long enough to be able to publish the message.
+- An mqtt-listener which is embedded in the mqtt_sub module and can control several pilight devices. Connection and subscription are kept alive "forever".
+
+.. _mqtt: 
+.. rubric:: mqtt
+
+.. code-block:: json
+   :linenos:
+
+   {
+     "hardware": {
+       "mqtt": {
+         "host": "localhost",
+         "port": 1883
+         "user": "username",
+         "pass": "password",
+         "listener_topic": "pilight/control/#,pilight/command/#"
```

Comment:
Isn't `topic` good enough?

Reply:
Also, was this the spot where you needed that JSON_ARRAY?

Reply:
No, this topic is only for the listener. All other clients must specify their own topic(s).  The other four settings, like "host" are relevant for all clients. Naming it this way it is more clear what the setting is meant for.

And no, I needed the json array for the genercic_label protocol to verify the value of the "blink" option.

Reply:
I am sorry, but the second answer above was a stupid answer. I mixed up two things. The answer is: Yes, that is where I needed that JSON_ARRY.